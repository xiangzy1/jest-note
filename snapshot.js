export const generateConfig = () => {
  return {
    server: 'http://localhost',
    port: '8080'
  }
}
export const generateTimeConfig = () => {
  return {
    time: new Date()
  }
}
