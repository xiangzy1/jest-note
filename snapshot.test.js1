import { generateConfig, generateTimeConfig } from "./snapshot";

test("test generateconfig", () => {
  // 测试配置文件，第一次生成快照，之后每次与快照对比
  expect(generateConfig()).toMatchSnapshot();
});
test("test generateconfigtime", () => {
  // time不必每次都相等
  expect(generateTimeConfig()).toMatchSnapshot({
    time: expect.any(Date),
  });
});
test("test generateconfigtime", () => {
  // 将snapshot放在行内
  expect(generateTimeConfig()).toMatchInlineSnapshot(
    {
      time: expect.any(Date),
    },
    `
    Object {
      "time": Any<Date>,
    }
  `
  );
});
