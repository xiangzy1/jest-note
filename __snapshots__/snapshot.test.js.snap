// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test generateconfig 1`] = `
Object {
  "port": "8080",
  "server": "http://localhost",
}
`;

exports[`test generateconfigtime 1`] = `
Object {
  "time": Any<Date>,
}
`;
